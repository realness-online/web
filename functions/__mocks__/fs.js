'use strict'
const fs = jest.genMockFromModule('fs')
const readFile_mock =  jest.fn((file, content_type, callback) => {
  // console.log(`file:${file}`, `content_type:${content_type}`)
  callback()
})
// fs.writeFileSync(locals.avatar, result.data)
const writeFileSync_mock = jest.fn((file, data) => {
  // console.log('inside writeFileSync_mock', file, data)
  return true
});
fs.readFile = readFile_mock
fs.writeFileSync = writeFileSync_mock
module.exports = fs
