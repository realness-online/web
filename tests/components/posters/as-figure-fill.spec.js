import { shallowMount } from '@vue/test-utils'
import as_fill_figure from '@/components/posters/as-figure-fill'
import get_item from '@/use/item'
import fs from 'fs'
const poster_html = fs.readFileSync(
  './__mocks__/html/poster.html',
  'utf8'
)
let poster = get_item(poster_html)
const itemid = '/+16282281824/posters/559666932867'
describe('@/components/posters/as-fill-figure.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(as_fill_figure, {
      props: { poster, itemid }
    })
  })
  describe('Renders', () => {
    it('a editor for a poster', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })
  describe.skip('Methods', () => {
    describe('#focus_on_active', () => {
      it('sets the focus on the active element', () => {
        wrapper.vm.focus_on_active()
      })
    })
    describe('#focus_on_active', () => {
      it('sets the color of active element', () => {
        wrapper.vm.set_input_color('bold')
      })
    })
  })
})
