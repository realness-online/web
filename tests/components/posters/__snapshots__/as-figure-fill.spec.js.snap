// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/posters/as-fill-figure.vue Renders a editor for a poster 1`] = `
<figure
  id="edit-fill"
  poster="[object Object]"
>
  <as-svg-stub
    immediate="true"
    itemid="/+16282281824/posters/559666932867"
    slice="false"
    tabable="true"
    tabindex="-1"
  />
  <figcaption>
    <input
      tabindex="-1"
      type="color"
    />
  </figcaption>
</figure>
`;
